### ISSUE TITLE
TypeError: 'str' object is not callable

### ISSUE THREAD

**Issue by smspillaz** [PyCQA/prospector#189]
This seems to happen when reading files on Python 3.5.0

```
Traceback (most recent call last):
  File "/Users/smspillaz/.python-django/lib/python3.5/site-packages/prospector-0.12.1-py3.5.egg/prospector/run.py", line 99, in execute
    messages = self.process_messages(found_files, messages)
  File "/Users/smspillaz/.python-django/lib/python3.5/site-packages/prospector-0.12.1-py3.5.egg/prospector/run.py", line 39, in process_messages
    return postfilter.filter_messages(filepaths, self.config.workdir, messages)
  File "/Users/smspillaz/.python-django/lib/python3.5/site-packages/prospector-0.12.1-py3.5.egg/prospector/postfilter.py", line 23, in filter_messages
    paths_to_ignore, lines_to_ignore, messages_to_ignore = get_suppressions(relative_filepaths, root, messages)
  File "/Users/smspillaz/.python-django/lib/python3.5/site-packages/prospector-0.12.1-py3.5.egg/prospector/suppression.py", line 96, in get_suppressions
    file_contents = encoding.read_py_file(abspath).split('\n')
  File "/Users/smspillaz/.python-django/lib/python3.5/site-packages/prospector-0.12.1-py3.5.egg/prospector/encoding.py", line 20, in read_py_file
    tokenize.detect_encoding(filepath)
  File "/Users/smspillaz/.python-django/lib/python3.5/tokenize.py", line 425, in detect_encoding
    first = read_or_stop()
  File "/Users/smspillaz/.python-django/lib/python3.5/tokenize.py", line 383, in read_or_stop
    return readline()
```

I had a look at the relevant code and it is not using [`tokenise`](https://docs.python.org/3/library/tokenize.html#tokenize.detect_encoding) correctly. It is passing the file path, instead of a readline function.

**Comment by smspillaz** [comment]
Fixed in #182 . Please have a look.

**Comment by carlio** [comment]
Closed as fix was merged
